react/components/ReactGooogleAutocomplete.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { RefObject } from 'react'
+import { Input, Spinner } from 'vtex.styleguide'
+import { injectIntl, InjectedIntlProps } from 'react-intl'
+
+import LocationInputIcon from './LocationInputIcon'
+
+/*
+
+Based on: https://github.com/ErrorPro/react-google-autocomplete/blob/master/src/index.js
+
+*/
+
+type AutocompletionRequest = google.maps.places.AutocompletionRequest
+interface Props {
+  isLoading: boolean
+  types?: AutocompletionRequest['types']
+  componentRestrictions?: AutocompletionRequest['componentRestrictions']
+  bounds?: AutocompletionRequest['bounds']
+  onPlaceSelected: (place: google.maps.places.PlaceResult) => void
+  value?: string
+  errorMessage?: string
+  onChange?: (e: any) => void
+  onSuffixPress: (e: any) => void
+  isFetchingPosition: boolean
+}
+
+class ReactGoogleAutocomplete extends React.Component<Props & InjectedIntlProps> {
+  input: RefObject<HTMLInputElement>
+  autocomplete: google.maps.places.Autocomplete | null
+  event: google.maps.MapsEventListener | null
+
+  constructor(props: Props & InjectedIntlProps) {
+    super(props)
+    this.input = React.createRef()
+    this.autocomplete = null
+    this.event = null
+  }
+
+  componentDidMount() {
+    if (!this.props.isLoading) {
+      this.setup()
+    }
+  }
+
+  setup = () => {
+    if (!this.input.current) {
+      return
+    }
+    const { types = ['(cities)'], componentRestrictions, bounds } = this.props
```

Comment:
o que é essa string `(cities)`?

Reply:
it's a default config, read: https://developers.google.com/places/supported_types#table3

Reply:
I pass `['address']`to work as we like

Reply:
I see. perhaps adding a comment with this link would be helpful
also perhaps set it as a `defaultProps` value, so that defaults are centralized

just a minor suggestion though, you can ignore it if you want